### ISSUE TITLE
[CRASH] Crash in drouting with high number of gateways on a carrier

### ISSUE THREAD

**Issue by bcnewlin** [OpenSIPS/opensips#1910]
**OpenSIPS version you are running**

```
version: opensips 2.4.6 (x86_64/linux)
flags: STATS: On, DISABLE_NAGLE, USE_MCAST, SHM_MMAP, PKG_MALLOC, QM_MALLOC, DBG_MALLOC, FAST_LOCK-ADAPTIVE_WAIT
ADAPTIVE_WAIT_LOOPS=1024, MAX_RECV_BUFFER_SIZE 262144, MAX_LISTEN 16, MAX_URI_SIZE 1024, BUF_SIZE 65535
poll method support: poll, epoll, sigio_rt, select.
git revision: aa89b8764
main.c compiled on 14:32:56 Nov 27 2019 with gcc 7
```

**Crash Core Dump**

https://pastebin.com/kQnwFvUn

**Describe the traffic that generated the bug**

High levels of normal SIP traffic.

**To Reproduce**

1. Configure a single dr_carrier with 100-150 gateways. We are not sure exactly how many were configured before the crashes started.
2. Start OpenSIPS
3. Start traffic
4. Check for crashes

**Relevant System Logs**

```
Nov 27 19:00:05 [19648] CRITICAL:core:sig_usr: segfault in process pid: 19648, id: 14
Nov 27 19:00:05 ip-10-32-213-45 kernel: opensips[19648]: segfault at 7ffd8d256000 ip 00007f39c7067ee2 sp 00007ffd8d24c1a0 error 4 in drouting.so[7f39c704b000+3a000]
```

**OS/environment information**
 - Operating System:  Amazon Linux 2
 - OpenSIPS installation:  git
 - other relevant information:

**Additional context**

**Comment by bogdan-iancu** [comment]
100% it is related to this buffer (with the size or 64) used for sorting the GWs - https://github.com/OpenSIPS/opensips/blob/6bd19fcc96b9a05c46b4c9bf319b500ed93a735e/modules/drouting/drouting.c#L2383

It is bad that there is no overflow check. As a test, you can increase the define to 200, recompile and see if it still crashes in your tests.

**Comment by bogdan-iancu** [comment]
@bcnewlin , could you please confirm the fix ? if everything ok I will do the backport to 2.4 and 3.0 also.

**Comment by bcnewlin** [comment]
@bogdan-iancu We are running on 2.4 only. Could you possibly provide the backported patch and I can test?

**Comment by bogdan-iancu** [comment]
Here is a 2.4 backport.
[opensips_2_4_drouting_1910.patch.txt](https://github.com/OpenSIPS/opensips/files/4029901/opensips_2_4_drouting_1910.patch.txt)
Let me know if it does the job.

**Comment by bcnewlin** [comment]
I have confirmed the fix with the patch. Thanks!

**Comment by bogdan-iancu** [comment]
backports done

**Comment by bcnewlin** [comment]
Thanks! Unfortunately when trying to pick up that backport I've run into an issue with the fix from 59400f7. :(

I've opened a separate ticket for that. #1932
